Add release notes for 0.9 Florence release (#254)
kyma-project/website: content/blog-posts/2019-04-08-release-notes-0.9/index.md

---
title: "Kyma 0.9 Florence"
author:
  name: "Klaudia Grzondziel, Technical Writer @Kyma"
tags:
  - release-notes
type: release
releaseTag: "0.9.0"
redirectFrom:
  - "/blog/release-notes-09"
---

> Write an introductory paragraph and present the most important release highlights from all components. List the highlights as bullet points and provide relative links to their corresponding sections.

<!-- overview -->

The highlights of Kyma 0.9 Florence include:

> - [{Feature or fix name}](#relative-link-to-subsection) - {One-sentence description}

Reviewer: ```suggestion
- [{Feature or fix name}](#relative-link-to-subsection) - {One-sentence test}
```

```yaml
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata:
  name: kubernetes-workloads
  labels:
    cms.kyma-project.io/view-context: docs-ui
    cms.kyma-project.io/order: "1"
    cms.kyma-project.io/group-name: Kubernetes
spec:
  displayName: "Workloads"
  description: "These are official Kubernetes docs about workloads"
  sources:
   - type: markdown
     name: markdown-files
     mode: package
     url: https://github.com/kubernetes/website/archive/master.zip
     filter: /content/en/docs/concepts/workloads/controllers
   - type: markdown
     name: markdown-files
     mode: package
     url: https://github.com/kubernetes/website/archive/master.zip
     filter: /content/en/docs/concepts/workloads/pods
```

### Service Catalog related views integrated with Headless CMS

> The Service Catalog-related views now read documentation provided with the new Headless CMS. This means that you can easily provide documentation and specifications for any service that you want to share through the Service Catalog. See the example DocsTopic custom resource for artificial Service Calls that would expose Slack APIs:

Reviewer 2: ```suggestion
The Service Catalog-related views now read documentation provided with the new Headless CMS. This means that you can easily provide documentation and specifications for any service that you want to share through the Service Catalog. See the example DocsTopic custom resource for artificial Service Class that would expose Slack APIs:
```

### Alerting and monitoring added

We added alerting and monitoring for eventing backing services.

## Installation

### Xip.io wildcard as the default DNS solution

The wildcard DNS provided by `xip.io` is integrated as the default DNS solution, which means you can now install Kyma on GKE and AKS clusters and create a playground environment even easier. Simply start the installation on your cluster and don't worry about owning a domain or configuring your cluster in a specific way. If you want to spin up a production-ready cluster, the process doesn't change. When you provide your own domain and TLS certificates, the Installer uses them instead of these provided by `xip.io`. Additionally, all scenarios support Application Connectivity.

### Test runner for the Kyma upgrade

We created a test runner and a place for end-to-end upgrade tests executed by [Kyma upgrade plan on CI](https://status.build.kyma-project.io/?job=post-master-kyma-gke-upgrade). The framework allows you to prepare the data and run tests against the prepared data. For more information on end-to-end upgrade tests, read [this](https://github.com/kyma-project/kyma/tree/master/tests/end-to-end/upgrade) document.

> ## Logging

Reviewer 3: The whole part about `Logging` must wait until 1.0

<!-- overview -->

The highlights of Kyma 0.9 Florence include:

- [Knative-based eventing by default](#kyma-with-knative-eventing) - The eventing mechanism is now based on Knative by default.
- [Xip.io wildcard as the default DNS solution](#installation) - We integrated the `xip.io` wildcard DNS as the default DNS solution.
- [Headless CMS component introduced for the Console UI](#headless-cms) - We introduced the Headless CMS component that allows you to store and manage content, and expose it through an API.
- [Updated Monitoring stack version](#monitoring-stack-version-update) - We updated the entire Monitoring stack.
- [Improved upgrade process](#test-runner-for-the-kyma-upgrade) - We developed a testing framework that automates checking the state of Kyma components after the upgrade.

See the overview of all changes in this release:

- [Application Connector](#application-connector) - Connector Service enhancements and extensions

> - [Console](#console) - Console views that display documentation use the new Headless CMS, new Log UI available in the Console UI

Reviewer 3: It looks like logging didn't make it to the release

```suggestion
- [Console](#console) - Console views that display documentation use the new Headless CMS.
```

### Headers management

We changed the Application Gateway proxy functionality and now the unnecessary headers, such as `X-Forwarded-For`, are removed while making calls to external solutions. For the full list of removed headers, read [this](https://kyma-project.io/docs/master/components/application-connector/#architecture-application-gateway-handling-of-headers) document.

## Console

### Documentation UI integrated with Headless CMS

The Documentation UI, which you can access in the Console UI, is now integrated with the Headless CMS. This means you can extend the Documentation view, the one you see after clicking the **?** icon, with your custom documentation topics as long as you have your docs written in Markdown and extended with additional metadata. The code snippet below is an example of a ClusterDocsTopic custom resource that will extend the Documentation UI navigation with a new element called **Workloads** which will be grouped under a new **Kubernetes** heading. After clicking **Workloads**, you will see the official Kubernetes documentation.  

> ```

Reviewer 4: ```yaml

> ```

Reviewer 2: please replace this snippet with 

```
cat <<EOF | kubectl apply -f -
apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata:
  labels:
    cms.kyma-project.io/view-context: docs-ui
    cms.kyma-project.io/group-name: components
    cms.kyma-project.io/order: "2"
  name: prometheus
spec:
  displayName: "Prometheus"
  description: "Some docs about Prometheus concepts"
  sources:
    - type: markdown
      name: docs
      mode: package
      url: https://github.com/prometheus/docs/archive/master.zip
      filter: content/docs/concepts
EOF
```

Reviewer 2: and replace sentence `with a new element called **Workloads** which will be grouped under a new **Kubernetes** heading` with `with a new element called **Prometheus** under a **Components** heading`

Reviewer 2: `After clicking **Workloads**, you will see the official Kubernetes documentation. ` change to `After clicking **Prometheus**, you will see the official Prometheus documentation. `

apiVersion: cms.kyma-project.io/v1alpha1
kind: ClusterDocsTopic
metadata:
  name: kubernetes-workloads
  labels:
    cms.kyma-project.io/view-context: docs-ui
    cms.kyma-project.io/order: "1"
    cms.kyma-project.io/group-name: Kubernetes
spec:
  displayName: "Workloads"
  description: "These are official Kubernetes docs about workloads"
  sources:
   - type: markdown
     name: markdown-files
     mode: package
     url: https://github.com/kubernetes/website/archive/master.zip
     filter: /content/en/docs/concepts/workloads/controllers
   - type: markdown
     name: markdown-files
     mode: package
     url: https://github.com/kubernetes/website/archive/master.zip
     filter: /content/en/docs/concepts/workloads/pods
```

### Service Catalog related views integrated with Headless CMS

> The Service Catalog-related views now read documentation provided with the new Headless CMS. This means that you can easily provide documentation and specifications for any service that you want to share through the Service Catalog. See the example DocsTopic custom resource that extends the Text Analytics service from Azure with additional documentation, Markdown docs, and the Swagger console based on the official Azure OpenAPI specification for the service:

Reviewer 2: ```suggestion
The Service Catalog-related views now read documentation provided with the new Headless CMS. This means that you can easily provide documentation and specifications for any service that you want to share through the Service Catalog. See the example DocsTopic custom resource for artificial Service Calls that would expose slack APIs:
```

> ```

Reviewer 4: ```yaml

apiVersion: cms.kyma-project.io/v1alpha1
kind: DocsTopic
metadata:

>   name: slack

Reviewer 2: ```suggestion
  name: {$SERVICE_CLASS_ID}
```

The Headless CMS itself does not yet deliver any customizable UI interface that could be used to publish a standalone documentation portal. Nevertheless, we already use it in the Console UI. For more details, read the Headless CMS [documentation](https://kyma-project.io/docs/components/headless-cms/).

### Asset Store supports a webhook service that can enhance the status of the CR with additional metadata for each file

You can now extend the Asset Store status sub-resource with additional metadata information for each file created by the controller. To do so, create a separate service that implements REST API and accepts `multipart/form-data`. One of the use cases for such a service is to extract the front matter metadata provided in any file of the `yaml` format.
See the example AssetStore resource that contains information about the status modification webhook:

> ```

Reviewer 4: ```yaml

<!-- overview -->

The highlights of Kyma 0.9 Florence include:

- [Knative-based eventing by default](#kyma-with-knative-eventing) - The eventing mechanism is now based on Knative by default.
- [Xip.io wildcard as the default DNS solution](#installation) - We integrated the `xip.io` wildcard DNS as the default DNS solution.
- [Increased security](#security) - We enabled mutual TLS by default.
- [Headless CMS component introduced for the Console UI](#headless-cms) - We introduced the Headless CMS component that allows you to store and manage content, and expose it through an API.
- [Updated Monitoring Stack version](#monitoring-stack-version-update) - We updated the entire Monitoring Stack.
- [Improved upgrade process](#test-runner-for-the-kyma-upgrade) - We developed a test framework that automates checking the state of Kyma components after upgrade.

See the overview of all changes in this release:

> - [Application Connector](#application-connector) - Connector Service enhancements and extensions, possibility to configure the expiration time of certificates separately

Reviewer 5: I want to remove the sentence `possibility to configure the expiration time of certificates separately`. It is only a small contribution.

Reviewer 6: done, removed everywhere
